alerter/alerter.cs

```diff
@@ -1,29 +1,42 @@
 ﻿using System;
+using System.Diagnostics;
 
-namespace AlerterSpace {
-    class Alerter {
+namespace AlerterSpace
+{
+    class Alerter
+    {
         static int alertFailureCount = 0;
-        static int networkAlertStub(float celcius) {
-            Console.WriteLine("ALERT: Temperature is {0} celcius", celcius);
+        static int networkAlertStub(float celsius)
+        {
+            Console.WriteLine("ALERT: Temperature is {0} celsius", celsius);
             // Return 200 for ok
             // Return 500 for not-ok
             // stub always succeeds and returns 200
+            if (celsius is <= 333 and >= 220)
+            {
+                return 500;
+            }
             return 200;
         }
-        static void alertInCelcius(float farenheit) {
-            float celcius = (farenheit - 32) * 5 / 9;
-            int returnCode = networkAlertStub(celcius);
-            if (returnCode != 200) {
+        static void alertIncelsius(float farenheit)
+        {
+            float celsius = (farenheit - 32) * 5 / 9;
+            int returnCode = networkAlertStub(celsius);

```

Comment:
production code should not be aware of test code or stubs. It should be abstracted by using interfaces or method delegates.